### ISSUE TITLE
How Siamese Layer and Regularizers play together

### ISSUE THREAD

**Issue by EderSantana** [fchollet/keras#1164]
Some layers are dependent on their regularizers to work correctly for example, the BatchNormalization Layer. How the Siamese layer plays with regularizers, does it collect the resulting regularizer for each input? 

cc @farizrahman4u
